coalib/settings/ConfigurationGathering.py

```diff
@@ -192,9 +193,11 @@ def load_configuration(arg_list, log_printer, arg_parser=None):
 
         sections = merge_section_dicts(sections, cli_sections)
 
-    for section in sections:
-        if section != 'default':
-            sections[section].defaults = sections['default']
+    for name, section in list(sections.items()):
+        section.set_default_section(sections)
+        if name == 'default':
+            logging.warning('Implicit \'Default\' section inheritance is '
+                            'deprecated. It will be removed soon.')
```

Comment:
at the very least we should check here if the section contains any important settings and allow the user to somehow get rid of this warning IMO. If this section isn't empty after your changes then your changes aren't complete yet are they?